components/base/kinematicbase/ptgKinematics.go

```diff
@@ -240,13 +243,15 @@ func (ptgk *ptgBaseKinematics) goToInputs(ctx context.Context, inputs []referenc
 				ptgk.inputLock.Unlock()
 				utils.SelectContextOrWait(ctx, time.Duration(inputUpdateStep*1000*1000)*time.Microsecond)
 			}
+			wg.Done()
 		})
 
 		if !utils.SelectContextOrWait(ctx, timestep) {
 			ptgk.logger.CDebug(ctx, ctx.Err().Error())
 			// context cancelled
 			break
 		}
+		wg.Wait()
```

Comment:
This should probably be in a defer under wg creation

Reply:
deferring `wg.Wait` causes for a data race. I will revert to as this was.

Reply:
Suppose we would like to run some `SetVelocity` call on a base for exactly 3.13 seconds.
We then enter 
```
utils.PanicCapturingGo(func() {
	// We need to update currentInputs as we move through the arc.
	// note inputUpdateStep == 0.1 seconds
	for timeElapsed := 0.; timeElapsed <= step.timestepSeconds; timeElapsed += inputUpdateStep {
		if ctx.Err() != nil {
			ptgk.logger.CDebug(ctx, ctx.Err().Error())
			// context cancelled
			break
		}
		distIncVel := step.linVelMMps.Y
		if distIncVel == 0 {
			distIncVel = step.angVelDegps.Z
		}
		ptgk.inputLock.Lock()
		ptgk.currentInput = []referenceframe.Input{inputs[0], inputs[1], {math.Abs(distIncVel) * timeElapsed}}
		ptgk.inputLock.Unlock()
		utils.SelectContextOrWait(ctx, time.Duration(inputUpdateStep*1000*1000)*time.Microsecond)
	}
})
```
since this is spawns a go-routine we proceed to 
```
if !utils.SelectContextOrWait(ctx, timestep) {
	ptgk.logger.CDebug(ctx, ctx.Err().Error())
	// context cancelled
	break
}
```
---------------------

Now, `!utils.SelectContextOrWait` will take `3.13` seconds, but the for loop within `PanicCapturingGo` will take slightly longer. The reason why it will take slightly longer is because we will be doing `3.13 / 0.1 = 31.3 => 32` iterations. Since each iteration lasts `0.1` seconds `PanicCapturingGo` will take exactly 3.2 seconds.

Hence, we need to explicitly specify `wg.Wait()` instead of deferring it.

Reply:
Looking at this more, I don't think there's a reason for that goroutine to exist. We could remove the goroutine, run that for loop in the main thread, and remove this problem entirely. No need for a waitgroup.

Reply:
sounds good. I also removed the `utils.SelectContextOrWait` conditional.